libbpf-tools/biotop.bpf.c

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright (c) 2022 Francis Laniel <flaniel@linux.microsoft.com>
+#include <vmlinux.h>
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_core_read.h>
+#include <bpf/bpf_tracing.h>
+
+#include "biotop.h"
+
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__uint(max_entries, 10240);
+	__type(key, struct request *);
+	__type(value, struct start_req_t);
+} start SEC(".maps");
+
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__uint(max_entries, 10240);
+	__type(key, struct request *);
+	__type(value, struct who_t);
+} whobyreq SEC(".maps");
+
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__uint(max_entries, 10240);
+	__type(key, struct info_t);
+	__type(value, struct val_t);
+} counts SEC(".maps");
+
+SEC("kprobe/blk_account_io_start")
```

Comment:
In the current kernel , the function(s) are inline [0] and instead `__blk_account_io_start` seems to be a good choice here. Do we still want to add this instead of `__blk_account_io_start` ?

[0]: [inline hot paths of blk_account_io_*()](https://lore.kernel.org/all/b0662a636bd4cc7b4f84c9d0a41efa46a688ef13.1633781740.git.asml.silence@gmail.com/)

`be6bfe36db17 block: inline hot paths of blk_account_io_*()`

Reply:
Hi.

I saw some of these functions were inlined into the kernel but I was not aware of how to probe only some functions (see my comment below).
I will add them and try to probe them only if their symbols exists (to filter out inlined functions).


Best regards.

Reply:
You can use:

https://github.com/iovisor/bcc/blob/8f40d6f57a8d94e7aee74ce358572d34d31b4ed4/libbpf-tools/trace_helpers.h#L20